dallinger/command_line/docker_ssh.py

```diff
@@ -778,20 +785,40 @@ def destroy(server, app):
     print(f"App {app} removed")
 
 
+def connect_ssh_client(client, host, user):
+    """Helper function to connect an SSH client with proper authentication."""
+    from dallinger.config import get_config
+
+    config = get_config()
+    config.load()
+    server_pem = config.get("server_pem", None)
+
+    if server_pem:
+        if not os.path.exists(server_pem):
+            raise FileNotFoundError(
+                f"SSH key file not found: {server_pem}\n"
+                "Please check that the path in your config file is correct.\n"
+                "You can set the path in either:\n"
+                "  - Your experiment's config.txt: server_pem = /path/to/key.pem\n"
+                "  - Your global config ~/.dallingerconfig: server_pem = /path/to/key.pem"
+            )
+        client.connect(host, username=user, key_filename=server_pem)
+    else:
+        client.connect(host, username=user)
+
+
 class Executor:
     """Execute remote commands using paramiko"""
 
     def __init__(self, host, user=None, app=None):
-        import paramiko
-
         self.app = app
         self.client = paramiko.SSHClient()
         # For convenience we always trust the remote host
         self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
         self.client.load_system_host_keys()
```

Comment:
Good idea to refactor. There is still some duplicated logic with `get_sftp` though:

```py
    client = paramiko.SSHClient()
    # For convenience we always trust the remote host
    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
    client.load_system_host_keys()
```

You could refactor that into `connect_ssh_client` too, perhaps? You could call it `get_connected_ssh_client`?

Relatedly, I don't think we are ever closing/disconnected these clients, but that's a topic for another issue I guess.